README.md

```diff
@@ -53,13 +53,39 @@ making use of the ROS cross compiled libraries and system dependencies. The exam
+Note that this approach doesn't allow building multi-target APKs; the ABI is fixed in [config.sh](https://github.com/Intermodalics/ros_android/blob/kinetic/scripts/config.sh) and the ABI filter in the Gradle buildscript should match it.
 - Finally, write your [bottom level CMakeLists file](https://github.com/Intermodalics/ros_android/blob/kinetic/example_workspace/src/hello_world_example_app/app/src/main/cpp/CMakeLists.txt). Note that the way of finding and adding packages as dependencies should be just as in any regular Catkin project.
 
 Once your project is ready to be built, place it inside a catkin workspace (i.e. `/path/to/your/workspace/src`). Then, it can be built with the standalone script like this:
 ```
 build_catkin_workspace.sh -w /path/to/your/workspace -p /path/to/install_space -e /path/to/workspace -b Debug -v 1
 ```
 
-where `/path/to/workspace` is the root directory of the cross compiled packages and libraries in the [installation step](#installation). Debug and verbose flags are recommended while developing.
\ No newline at end of file
+where `/path/to/workspace` is the root directory of the cross compiled packages and libraries in the [installation step](#installation). Debug and verbose flags are recommended while developing.
+
+### Handling plugins
+
+Unlike regular ROS development under Linux, plugins are compiled as static libraries and linked to other libraries or applications that make use of them. The patched version of pluginlib does not load shared objects in the filesystem dynamically during runtime; all the plugins should be linked to your own shared libraries or applications at compile time, ensuring that the required symbols are not stripped by the linker.
+
+The `nodelet` sample application [CMakeLists](https://github.com/Intermodalics/ros_android/blob/kinetic/example_workspace/src/nodelet_sample_app/jni/CMakeLists.txt) is a good example to start with; note that `-Wl, --whole-archive` flags are used for plugins in particular to prevent the linker from stripping symbols that need to be included. In general terms, a structure like this one should work:
+
+```
+find_package(catkin REQUIRED COMPONENTS plugin_package_name)	# Add all required catkin components as well.
+
+find_library(plugin_name_PLUGIN NAMES plugin_name) # List all the required plugins.
+
+SET (WHOLE_ARCHIVE_LIBRARIES
+     -Wl,--whole-archive
+     ${plugin_name_PLUGIN}	# Add all the required plugins.
+     -Wl,--no-whole-archive
+)
+
+# add lib dependencies
+target_link_libraries(${PROJECT_NAME}
+	# Add app required libraries e.g. android here.
+    ${catkin_LIBRARIES}
+    ${WHOLE_ARCHIVE_LIBRARIES}
```

Comment:
Swap `${catkin_LIBRARIES}` and `${WHOLE_ARCHIVE_LIBRARIES}`?

That is what we typically do and what is in the examples. At least for executable targets the order does matter because the linker only adds symbols from static libraries listed in `${catkin_LIBRARIES}` which have been undefined before. As an example, if one of the plugins depends on a certain function of `roscpp` that is used nowhere else, linking the plugin library after `${catkin_LIBRARIES}` will result in a linker error because the symbol has been discarded when linking `libroscpp.a`.

Reply:
Addressed in https://github.com/Intermodalics/ros_android/pull/111/commits/9614181087442a33da680162364d97f8fe59c612.